examples/src/keyring/aws_kms/custom_client_supplier.py

```diff
@@ -0,0 +1,115 @@
+if you need different credentials to talk to different AWS regions.
+This might be because you are crossing partitions (ex: ``aws`` and ``aws-cn``)
+or if you are working with regions that have separate authentication silos
+like ``ap-east-1`` and ``me-south-1``.
+
+This example shows how to create a client supplier
+that will supply KMS clients with valid credentials for the target region
+even when working with regions that need different credentials.
+
+https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/choose-keyring.html#use-kms-keyring
+
+For an example of how to use the KMS keyring with CMKs in multiple regions,
+see the ``keyring/aws_kms/multiple_regions`` example.
+
+For another example of how to use the KMS keyring with a custom client configuration,
+see the ``keyring/aws_kms/custom_kms_client_config`` example.
+
+For examples of how to use the KMS keyring in discovery mode on decrypt,
+see the ``keyring/aws_kms/discovery_decrypt``,
+``keyring/aws_kms/discovery_decrypt_in_region_only``,
+and ``keyring/aws_kms/discovery_decrypt_with_preferred_region`` examples.
+"""
+from botocore.client import BaseClient
+from botocore.session import Session
+
+import aws_encryption_sdk
+from aws_encryption_sdk.keyrings.aws_kms import KmsKeyring
+from aws_encryption_sdk.keyrings.aws_kms.client_suppliers import ClientSupplier, DefaultClientSupplier
+
+try:  # Python 3.5.0 and 3.5.1 have incompatible typing modules
+    from typing import Union  # noqa pylint: disable=unused-import
+except ImportError:  # pragma: no cover
+    # We only actually need these imports when running the mypy checks
+    pass
+
+
+class MultiPartitionClientSupplier(ClientSupplier):
+    """Client supplier that supplies clients across AWS partitions and identity silos."""
+
+    def __init__(self):
+        """Set up default client suppliers for identity silos."""
+        self._china_supplier = DefaultClientSupplier(botocore_session=Session(profile="china"))
+        self._middle_east_supplier = DefaultClientSupplier(botocore_session=Session(profile="middle-east"))
+        self._hong_kong_supplier = DefaultClientSupplier(botocore_session=Session(profile="hong-kong"))
+        self._default_supplier = DefaultClientSupplier()
+
+    def __call__(self, region_name):
+        # type: (Union[None, str]) -> BaseClient
+        """Return a client for the requested region.
+
+        :rtype: BaseClient
+        """
+        if region_name.startswith("cn-"):
+            return self._china_supplier(region_name)
+
+        if region_name.startswith("me-"):
+            return self._middle_east_supplier(region_name)
+
+        if region_name == "ap-east-1":
+            return self._hong_kong_supplier(region_name)
+
+        return self._default_supplier(region_name)
+
+
+def run(aws_kms_cmk, source_plaintext):
+    # type: (str, bytes) -> None
+    """Demonstrate an encrypt/decrypt cycle using a KMS keyring with a custom client supplier.
```

Comment:
```suggestion
    """Demonstrate an encrypt/decrypt cycle using a KMS keyring with a custom client supplier.
```
I ran into the same dilemma in the docs. Does the "custom" before client supplier imply a configured standard client supplier, or a different implementation of the client supplier interface?

Reply:
I would assume that "custom" means "something that is not provided out of the box". A configuration of a built-in feature does not say "custom" to me